Functions/texturl.cs

```diff
@@ -0,0 +1,232 @@
+                        .AddEnvironmentVariables()
+                        .Build();
+                */
+
+                //access azure keyvault
+                var serviceTokenProvider = new AzureServiceTokenProvider();
+                log.LogInformation("serviceTokenProvider");
+                var keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(serviceTokenProvider.KeyVaultTokenCallback));
+                log.LogInformation("keyVaultClient");
+
+                //SecretBundle secretValues;
+                String uri = String.Empty;
+                String key = String.Empty;
+                String database = String.Empty;
+                String collection = String.Empty;
+
+                try
+                {
+                    SecretBundle secretURI = await keyVaultClient.GetSecretAsync("https://francesco-key-vault.vault.azure.net/secrets/cosmos-uri/");
+                    SecretBundle secretKey = await keyVaultClient.GetSecretAsync("https://francesco-key-vault.vault.azure.net/secrets/cosmos-key/");
+                    SecretBundle secretDB = await keyVaultClient.GetSecretAsync("https://francesco-key-vault.vault.azure.net/secrets/cosmos-db-name/");
+                    SecretBundle secretTable = await keyVaultClient.GetSecretAsync("https://francesco-key-vault.vault.azure.net/secrets/cosmos-table/");
+
+                    uri = secretURI.Value;
+                    key = secretKey.Value;
+                    database = secretDB.Value;
+                    collection = secretTable.Value;
+                    log.LogInformation("Secret Values retrieved from KeyVault.");
+                }
+                catch (Exception kex)
+                {
+                    return (ObjectResult)new ObjectResult(kex.Message.ToString());
+                }
+
+                //declare client
+                DocumentClient dbClient = new DocumentClient(new Uri(uri), key);
+                log.LogInformation("new DocumentClient");
+
+                try
+                {
+                    var collectionUri = UriFactory.CreateDocumentCollectionUri(database, collection);
+                    var query = "SELECT * FROM Books b WHERE b.id=\"" + bookid + "\"";
+                    var crossPartition = new FeedOptions { EnableCrossPartitionQuery = true };
+                    var documents = dbClient.CreateDocumentQuery(collectionUri, query, crossPartition).ToList();
+                    log.LogInformation($"document retrieved -> {documents.Count().ToString()}");
+
+                    Book b = documents.ElementAt(0);
+                    //update
+                    b.Pages.ElementAt(int.Parse(pageid) - 1).Languages.ElementAt(languagecode.Equals("en_US") ? 0 : 1).Text_Url = data.text.ToString();
+                    var result = await dbClient.UpsertDocumentAsync(UriFactory.CreateDocumentCollectionUri(database, collection), b);
```

Comment:
im not seeing where you are checking if the book/page/language already exists before you post